### ISSUE TITLE
THE DEAD NEURON PROBLEM

### ISSUE THREAD

**Issue by ZackWave** [BindsNET/bindsnet#613]
With the training, almost all neurons can't be activated.
The neural network has about 500 neurons and is divided into 8 communities. 
At the end of training, only the part connected with the input can send out pulses.

**Comment by Hananel-Hazan** [comment]
Thanks for using BindsNET. I'm not sure what the issue you facing.

**Comment by ZackWave** [comment]
I used bindnet to implement this network. After using postpre, only the part connected to input can send pulses, and the other parts will not work.

**Comment by ZackWave** [comment]
> 感谢您使用 BindsNET。这是 BindsNET 的问题，还是您有什么要讨论的更适合讨论部分？

I used bindnet to implement this network. After using postprep, only the part connected to input can send pulses, and the other parts will not work.

**Comment by ZackWave** [comment]
> 感谢您使用 BindsNET。这是 BindsNET 的问题，还是您有什么要讨论的更适合讨论部分？

I used bindsnet to implement this network. After using postpre, only the part connected to input can send pulses, and the other parts will not work.

**Comment by Hananel-Hazan** [comment]
What network did you implement?

**Comment by ZackWave** [comment]
> 您实施了什么网络？

Network(
  (inpt_layer): Input()
  (M_layer): IzhikevichNodes()
  (output_layer): IzhikevichNodes()
  (inpt_layer_to_M_layer): Connection(
    (source): Input()
    (target): IzhikevichNodes()
  )
  (M_layer_to_M_layer): Connection(
    (source): IzhikevichNodes()
    (target): IzhikevichNodes()
  )
  (M_layer_to_output_layer): Connection(
    (source): IzhikevichNodes()
    (target): IzhikevichNodes()
  )
)

**Comment by ZackWave** [comment]
> 您实施了什么网络？

input_data = poisson(datum=frame, time=time, device=device)
or
input_data = bernoulli(datum=frame, time=time, device=device)

**Comment by ZackWave** [comment]
> What network did you implement?

This neural network can learn with Bernoulli but not with Poisson. It doesn't work even if made deeper. The learning rate is 1e-2.

**Comment by Hananel-Hazan** [comment]
I'm sorry, I dont understand what code do you use, and what is the issue you facing. From the little information that I have, you may have you may need to use higher weights values. But that is a guess.

**Comment by ZackWave** [comment]
> 对不起，我不明白你用什么代码，你面临的问题是什么。从我掌握的少量信息来看，您可能需要使用更高的权重值。但这只是猜测。

Thank you, the adjustment of wmax was effective.

**Comment by ZackWave** [comment]
> 对不起，我不明白你用什么代码，你面临的问题是什么。从我掌握的少量信息来看，您可能需要使用更高的权重值。但这只是猜测。

How should I set appropriate wmax, wmin, and norm values? And how can I achieve sparse connections? I have tried setting some weights to zero, but this method does not work due to postpre.

**Comment by Hananel-Hazan** [comment]
Some parameters like wmax, wmin and others needed to be tune by trial and error. About the sparse connection, currently PyTorch dont have full support for sparse matrices, therefore we cant use sparse connections in BindsNET. Hopefully the situation will improve in the future.

**Comment by ZackWave** [comment]
> 一些参数，如wmax，wmin和其他参数需要通过反复试验进行调整。关于稀疏连接，目前 PyTorch 没有完全支持稀疏矩阵，因此我们不能在 BindsNET 中使用稀疏连接。希望将来情况会有所改善。

How to implement STDP learning rule on inhibitory neurons.

**Comment by Hananel-Hazan** [comment]
The same as you implement the rule on excitation neurons just with negative weights

**Comment by ZackWave** [comment]
> 就像你只用负权重在激发神经元上实现规则一样

https://snntorch.readthedocs.io/en/latest/tutorials/tutorial_5.html

**Comment by Hananel-Hazan** [comment]
This is a nice paper but not biological realistic, since synapse polarity cant change in biology and BP can flip the sign. In BindsNET you can define the synapse in such a way that this event never happen. Please see the BindsNET documentation.
